src/librustdoc/html/static/js/storage.js

```diff
@@ -76,6 +82,16 @@ function onEach(arr, func, reversed) {
     return false;
 }
 
+/**
+ * Turn an HTMLCollection or a NodeList into an Array, then run a callback
+ * for every element. This is useful because iterating over an HTMLCollection
+ * or a "live" NodeList while modifying it can be very slow.
+ * https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection
+ * https://developer.mozilla.org/en-US/docs/Web/API/NodeList
+ * @param {NodeList<?>|HTMLCollection<?>} lazyArray  - An array to iterate over
+ * @param {function(?)}                   func       - The callback
+ * @param {boolean}                       [reversed] - whether to iterate in reverse
```

Comment:
Why is `[reversed]` between square brackets? Also, the doc part doesn't start with a capital letter unlike the other lines (not sure if important or not though).

Reply:
The brackets indicate an optional parameter in JSDoc syntax. Note [there are two ways to indicate optionality](https://jsdoc.app/tags-param.html#optional-parameters-and-default-values). This one, and:

```
 * @param {boolean=} reversed - Whether to iterate in reverse
```

The version with = is specific to Closure Compiler. I like the bracketed version better.

Reply:
Oh I see. I agree. Weird syntax. Maybe we shouldn't allow optional parameters.